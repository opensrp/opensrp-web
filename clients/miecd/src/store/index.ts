import { connectReducer, getConnectedStore } from '@onaio/connected-reducer-registry';
import { gateKeeperReducer, gateKeeperReducerName } from '@onaio/gatekeeper';
import reducerRegistry, { combine, Registry } from '@onaio/redux-reducer-registry';
import session, { reducerName as sessionReducer } from '@onaio/session-reducer';

/** Initial reducers in the reducer registry */
const defaultReducers: Registry = {
    router: connectReducer as any /** Dirty hack because connectRouter LocationChangeAction does not extend Redux.AnyAction */,
};

/** Add users reducer to registry */
defaultReducers[sessionReducer] = session;
defaultReducers[gateKeeperReducerName] = gateKeeperReducer;

/** The initial store for the reveal web app */
const store = getConnectedStore(defaultReducers);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener(reducers => {
    store.replaceReducer(combine(reducers));
});

export default store;
