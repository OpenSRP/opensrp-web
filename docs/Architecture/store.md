# overview

This documentation will take you through the process of creating and working with:

    - reducers
    - actions
    - selectors

## Where do I find the store

the store reducer modules are found within src/store. The initial directory structure looks something like this: (Note: this structure only depicts the initial directory structure and will as expected contain more directories and files as more are added with time)

```
...
├── src
│   ├── ...
│   ├── store
│   │   ├── ducks
│   │   │    ├── superset.ts
│   │   │    ├── tests/
│   │   ├── tests
│   │   ├── index.ts
│   │   └── selectors.ts
...
```

A single reducer module will contain the afore mentioned parts i.e. its respective actions, action creators if any, its selectors and the reducer function

## Actions

An action can be any object with a type property, they are the only way provisioned for changing the state in a redux store.

### Adding Actions

Its a common convention as well as recommended one to Use type constants, [read more on this here]().

steps:

- Add An interface describing the action structure: redux exposes an `AnyAction` type that can be extended to write custom action interfaces
- Create a general actiontypes to be used to type check all actions that will be passed to the reducer function as an argument. this will be in the form of [intersection type]()

## Selectors

whereas the action puts data in the store; the selectors on the other hand act as getters of the slice of data that this reducer module handles

selectors are functions that will take at least one argument, the store, Then based on any other arguments like say an id, can be used to filter down the data retrieved.

## The Reducer

The reducer function, takes 2 arguments; the state and an action.
using the type property of the action, the reducer should then know how to modify the state.
such functionality could be implemented as a form of switch strcture such as a list of if statements or the more recommended approach which is to use the actual switch statement

### Adding the Reducer

- Define an interface for the data to be stored in the store
- Define an interface for the store state.
- write the actual reducer function, while applying the required type annotations as required.
