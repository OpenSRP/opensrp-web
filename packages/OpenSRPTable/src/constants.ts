// string literals
export const PLEASE_DEFINE_SOME_COLUMNS = 'Please define some columns';
export type PLEASE_DEFINE_SOME_COLUMNS = typeof PLEASE_DEFINE_SOME_COLUMNS;
