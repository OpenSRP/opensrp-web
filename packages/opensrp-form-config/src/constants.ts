export const MAKE_RELEASE_LABEL = 'Make Release';
export const FILE_NAME_LABEL = 'File Name';
export const FILE_VERSION_LABEL = 'File Version';
export const IDENTIFIER_LABEL = 'Identifier';
export const MODULE_LABEL = 'Module';
export const EDIT_LABEL = 'Edit';
export const DOWNLOAD_LABEL = 'Download';
export const UPLOAD_EDIT_LABEL = 'Upload edit';
