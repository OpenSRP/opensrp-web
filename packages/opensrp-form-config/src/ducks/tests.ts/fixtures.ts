export const fixManifestReleases = [
    {
        identifier: '1.0.11',
        json: '{"forms_version":"1.0.11","identifiers":["blood_screening.json"}',
        appId: 'org.smartregister.reveal',
        appVersion: '1.2.14',
        createdAt: '2020-06-08T15:00:38.413+02:00',
        updatedAt: '2020-06-08T15:00:38.413+02:00',
    },
    {
        identifier: '1.0.12',
        json: '{"forms_version":"1.0.12","identifiers":["blood_screening.json"}',
        appId: 'org.smartregister.reveal',
        appVersion: '1.2.14',
        createdAt: '2020-06-08T15:03:00.950+02:00',
        updatedAt: '2020-06-08T15:03:00.950+02:00',
    },
    {
        identifier: '1.0.16',
        json: '{"forms_version":"1.0.16","identifiers":["blood_screening.json"}',
        appId: 'org.smartregister.reveal',
        appVersion: '1.2.16',
        createdAt: '2020-06-09T14:40:04.189+02:00',
        updatedAt: '2020-06-09T14:40:04.189+02:00',
    },
];
