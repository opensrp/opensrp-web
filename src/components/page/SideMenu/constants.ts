import { ModulePageLink, NavigationModule, PageLink } from './SubMenu';

import {
  ADMIN,
  ALL_CLIENTS,
  ANALYSIS,
  ANALYSIS_URL,
  ANC,
  ANC_URL,
  CHILD,
  CHILD_URL,
  CLIENT_RECORDS,
  CLIENT_URL,
  COMPARTMENTS,
  COMPARTMENTS_URL,
  HOME,
  HOME_URL,
  HOUSEHOLD,
  HOUSEHOLD_URL,
  LOCATIONS,
  LOCATIONS_URL,
  LOGFACE,
  LOGFACE_URL,
  NBC_AND_PNC,
  NBC_AND_PNC_URL,
  NUTRITION,
  NUTRITION_URL,
  PREGNANCY,
  PREGNANCY_URL,
  REPORTS,
  REPORTS_URL,
  ROLE_URL,
  ROLES,
  TEAM_URL,
  TEAMS,
  USER_URL,
  USERS,
} from '../../../constants';

import { IconProp } from '@fortawesome/fontawesome-svg-core';

import {
  ENABLE_ANC_PAGE,
  ENABLE_CHILD_PAGE,
  ENABLE_CLIENT_PAGE,
  ENABLE_HOUSEHOLD_PAGE,
  ENABLE_LOCATION_PAGE,
  ENABLE_ROLE_PAGE,
  ENABLE_TEAM_PAGE,
  ENABLE_USER_PAGE,
} from '../../../configs/env';

// Page links

export const LOG_FACE: PageLink = {
  label: 'logface',
  url: '/logface',
};

export const CompartMents: PageLink = {
  label: 'compartments',
  url: '/compartments',
};

export const Analysis: PageLink = {
  label: 'Analysis',
  url: '/analysis',
};
export const CLIENT_PAGE_NAVIGATION: PageLink = {
  label: ALL_CLIENTS,
  url: CLIENT_URL,
};
export const HOUSEHOLD_PAGE_NAVIGATION: PageLink = {
  label: HOUSEHOLD,
  url: HOUSEHOLD_URL,
};
export const ANC_PAGE_NAVIGATION: PageLink = {
  label: ANC,
  url: ANC_URL,
};
export const CHILD_PAGE_NAVIGATION: PageLink = {
  label: CHILD,
  url: CHILD_URL,
};

export const USERS_PAGE_NAVIGATION: PageLink = {
  label: USERS,
  url: USER_URL,
};
export const ROLES_PAGE_NAVIGATION: PageLink = {
  label: ROLES,
  url: ROLE_URL,
};
export const TEAMS_PAGE_NAVIGATION: PageLink = {
  label: TEAMS,
  url: TEAM_URL,
};
export const LOCATIONS_PAGE_NAVIGATION: PageLink = {
  label: LOCATIONS,
  url: LOCATIONS_URL,
};

export const PREGNANCY_PAGE_NAVIGATION: PageLink = {
  label: PREGNANCY,
  url: PREGNANCY_URL,
};

export const NUTRITION_PAGE_NAVIGATION: PageLink = {
  label: NUTRITION,
  url: NUTRITION_URL,
};

export const NEWBORN_AND_POSTNATAL_PAGE_NAVIGATION: PageLink = {
  label: NBC_AND_PNC,
  url: NBC_AND_PNC_URL,
};

export const COMPARTMENTS_PAGE_NAVIGATION: PageLink = {
  label: COMPARTMENTS,
  url: COMPARTMENTS_URL,
};

export const LOGFACE_PAGE_NAVIGATION: PageLink = {
  label: LOGFACE,
  url: LOGFACE_URL,
};

export const ANALYSIS_PAGE_NAVIGATION: PageLink = {
  label: ANALYSIS,
  url: ANALYSIS_URL,
};

// icons
export const homeNavIcon: IconProp = ['fas', 'home'];
export const pregancyNavIcon: IconProp = ['far', 'user'];
export const nbcandpncNavIcon: IconProp = ['far', 'user'];
export const nutritionNavIcon: IconProp = ['far', 'user'];
export const newbornAndPostnatalNavIcon: IconProp = ['far', 'user'];
export const clientModuleNavIcon: IconProp = ['far', 'user'];
export const reportModuleNavIcon: IconProp = ['fas', 'chart-line'];
export const adminModuleNavIcon: IconProp = ['fas', 'cog'];

// Navigation links for navigation module. collapses when clicked to display the pageLinks

export const HOME_PARENT_NAV: ModulePageLink = {
  icon: homeNavIcon,
  label: HOME,
  url: HOME_URL,
};

export const NBC_AND_PNC_PARENT_NAV: ModulePageLink = {
  icon: nbcandpncNavIcon,
  label: NBC_AND_PNC,
  url: NBC_AND_PNC_URL,
};

export const NUTRITION_PARENT_NAVIGATION: ModulePageLink = {
  icon: nutritionNavIcon,
  label: NUTRITION,
  url: NUTRITION_URL,
};

export const PREGNANCY_MODULE_PARENT_NAV: ModulePageLink = {
  icon: pregancyNavIcon,
  label: PREGNANCY,
  url: PREGNANCY_URL,
};

export const CLIENT_MODULE_PARENT_NAV: ModulePageLink = {
  icon: clientModuleNavIcon,
  label: CLIENT_RECORDS,
};
export const REPORT_MODULE_PARENT_NAV: ModulePageLink = {
  icon: reportModuleNavIcon,
  label: REPORTS,
  url: REPORTS_URL,
};
export const ADMIN_MODULE_PARENT_NAV: ModulePageLink = {
  icon: adminModuleNavIcon,
  label: ADMIN,
};

// nav module objects
export const HOME_NAVIGATION_MODULE: NavigationModule = {
  childNavs: [],
  parentNav: HOME_PARENT_NAV,
};

export const PREGNANCY_NAVIGATION_MODULE: NavigationModule = {
  childNavs: [
    LOGFACE_PAGE_NAVIGATION,
    COMPARTMENTS_PAGE_NAVIGATION,
    ANALYSIS_PAGE_NAVIGATION,
  ].filter((childNav): childNav is PageLink => typeof childNav !== 'boolean'),
  parentNav: PREGNANCY_MODULE_PARENT_NAV,
};

export const NBC_AND_PNC_NAVIGATION_MODULE: NavigationModule = {
  childNavs: [LOGFACE_PAGE_NAVIGATION],
  parentNav: NBC_AND_PNC_PARENT_NAV,
};

export const NUTRITION_MODULE: NavigationModule = {
  childNavs: [LOGFACE_PAGE_NAVIGATION],
  parentNav: NUTRITION_PARENT_NAVIGATION,
};

export const CLIENT_NAVIGATION_MODULE: NavigationModule = {
  childNavs: [
    ENABLE_CLIENT_PAGE && CLIENT_PAGE_NAVIGATION,
    ENABLE_HOUSEHOLD_PAGE && HOUSEHOLD_PAGE_NAVIGATION,
    ENABLE_ANC_PAGE && ANC_PAGE_NAVIGATION,
    ENABLE_CHILD_PAGE && CHILD_PAGE_NAVIGATION,
  ].filter((childNav): childNav is PageLink => typeof childNav !== 'boolean'),
  parentNav: CLIENT_MODULE_PARENT_NAV,
};

export const REPORT_NAVIGATION_MODULE: NavigationModule = {
  childNavs: [],
  parentNav: REPORT_MODULE_PARENT_NAV,
};
export const ADMIN_NAVIGATION_MODULE: NavigationModule = {
  childNavs: [
    ENABLE_USER_PAGE && USERS_PAGE_NAVIGATION,
    ENABLE_ROLE_PAGE && ROLES_PAGE_NAVIGATION,
    ENABLE_TEAM_PAGE && TEAMS_PAGE_NAVIGATION,
    ENABLE_LOCATION_PAGE && LOCATIONS_PAGE_NAVIGATION,
  ].filter<PageLink>(
    (childNav: PageLink | boolean): childNav is PageLink => typeof childNav !== 'boolean'
  ),
  parentNav: ADMIN_MODULE_PARENT_NAV,
};

/** Enable Clients Module from its child pages */
export const ENABLE_CLIENT_RECORDS_MODULE =
  ENABLE_ANC_PAGE || ENABLE_CLIENT_PAGE || ENABLE_HOUSEHOLD_PAGE || ENABLE_HOUSEHOLD_PAGE;
export type ENABLE_CLIENT_RECORDS_MODULE = typeof ENABLE_CLIENT_RECORDS_MODULE;

/** Enable Admin Module from its child pages */
export const ENABLE_ADMIN_MODULE =
  ENABLE_USER_PAGE || ENABLE_ROLE_PAGE || ENABLE_TEAM_PAGE || ENABLE_LOCATION_PAGE;
export type ENABLE_ADMIN_MODULE = typeof ENABLE_ADMIN_MODULE;
