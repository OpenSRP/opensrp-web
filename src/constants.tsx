import { IconProp } from '@fortawesome/fontawesome-svg-core';
import { NavModuleObj } from './components/page/SideMenu/SideMenu';
import { NavCollapseObj, NavObj } from './components/page/SubMenu/SubMenu';

// internal urls
export const LOGIN_URL = '/login';
export type LOGIN_URL = typeof LOGIN_URL;
export const LOGOUT_URL = '/logout';
export type LOGOUT_URL = typeof LOGOUT_URL;
export const HOME_URL = '/';
export type HOME_URL = typeof HOME_URL;

export const CLIENT_URL = '/clients';
export type CLIENT_URL = typeof CLIENT_URL;
export const HOUSEHOLD_URL = '/404';
export type HOUSEHOLD_URL = typeof HOUSEHOLD_URL;
export const ANC_URL = '/404';
export type ANC_URL = typeof ANC_URL;
export const CHILD_URL = '/404';
export type CHILD_URL = typeof CHILD_URL;

export const USER_URL = '/404';
export type USER_URL = typeof USER_URL;
export const ROLE_URL = '/404';
export type ROLE_URL = typeof ROLE_URL;
export const TEAM_URL = '/404';
export type TEAM_URL = typeof TEAM_URL;
export const LOCATION_URL = '/404';
export type LOCATION_URL = typeof LOCATION_URL;

// side nav confs
// identifiers
export const CLIENT_COLLAPSE_NAVIGATION_IDENTIFIER = 'CLIENT';
export type CLIENT_COLLAPSE_NAVIGATION_IDENTIFIER = typeof CLIENT_COLLAPSE_NAVIGATION_IDENTIFIER;
export const REPORT_COLLAPSE_NAVIGATION_IDENTIFIER = 'REPORT';
export type REPORT_COLLAPSE_NAVIGATION_IDENTIFIER = typeof REPORT_COLLAPSE_NAVIGATION_IDENTIFIER;
export const ADMIN_COLLAPSE_NAVIGATION_IDENTIFIER = 'ADMIN';
export type ADMIN_COLLAPSE_NAVIGATION_IDENTIFIER = typeof ADMIN_COLLAPSE_NAVIGATION_IDENTIFIER;

// collapse labels
export const CLIENT_COLLAPSE_NAVIGATION_LABEL = 'Client Records';
export type CLIENT_COLLAPSE_NAVIGATION_LABEL = typeof CLIENT_COLLAPSE_NAVIGATION_LABEL;
export const REPORT_COLLAPSE_NAVIGATION_LABEL = 'Reports';
export type REPORT_COLLAPSE_NAVIGATION_LABEL = typeof REPORT_COLLAPSE_NAVIGATION_LABEL;
export const ADMIN_COLLAPSE_NAVIGATION_LABEL = 'Admin';
export type ADMIN_COLLAPSE_NAVIGATION_LABEL = typeof ADMIN_COLLAPSE_NAVIGATION_LABEL;

// icons
export const CLIENT_COLLAPSE_NAVIGATION_ICON: IconProp = ['far', 'user'];
export const REPORT_COLLAPSE_NAVIGATION_ICON: IconProp = ['fas', 'chart-line'];
export const ADMIN_COLLAPSE_NAVIGATION_ICON: IconProp = ['fas', 'cog'];

// nested labels
export const CLIENT_PAGE_NAVIGATION_LABEL = 'All Clients';
export type CLIENT_PAGE_NAVIGATION_LABEL = typeof CLIENT_PAGE_NAVIGATION_LABEL;
export const HOUSEHOLD_PAGE_NAVIGATION_LABEL = 'Household';
export type HOUSEHOLD_PAGE_NAVIGATION_LABEL = typeof HOUSEHOLD_PAGE_NAVIGATION_LABEL;
export const ANC_PAGE_NAVIGATION_LABEL = 'ANC';
export type ANC_PAGE_NAVIGATION_LABEL = typeof ANC_PAGE_NAVIGATION_LABEL;
export const CHILD_PAGE_NAVIGATION_LABEL = 'Child';
export type CHILD_PAGE_NAVIGATION_LABEL = typeof CHILD_PAGE_NAVIGATION_LABEL;

export const USER_PAGE_NAVIGATION_LABEL = 'Users';
export type USER_PAGE_NAVIGATION_LABEL = typeof USER_PAGE_NAVIGATION_LABEL;
export const ROLE_PAGE_NAVIGATION_LABEL = 'Roles';
export type ROLE_PAGE_NAVIGATION_LABEL = typeof ROLE_PAGE_NAVIGATION_LABEL;
export const TEAM_PAGE_NAVIGATION_LABEL = 'Teams';
export type TEAM_PAGE_NAVIGATION_LABEL = typeof TEAM_PAGE_NAVIGATION_LABEL;
export const LOCATION_PAGE_NAVIGATION_LABEL = 'Locations';
export type LOCATION_PAGE_NAVIGATION_LABEL = typeof LOCATION_PAGE_NAVIGATION_LABEL;

// nav objects
// client page
export const CLIENT_PAGE_NAVIGATION_OBJECT: NavObj = {
  navLabel: CLIENT_PAGE_NAVIGATION_LABEL,
  navURL: CLIENT_URL,
};
export const HOUSEHOLD_PAGE_NAVIGATION_OBJECT: NavObj = {
  navLabel: HOUSEHOLD_PAGE_NAVIGATION_LABEL,
  navURL: HOUSEHOLD_URL,
};
export const ANC_PAGE_NAVIGATION_OBJECT: NavObj = {
  navLabel: ANC_PAGE_NAVIGATION_LABEL,
  navURL: ANC_URL,
};
export const CHILD_PAGE_NAVIGATION_OBJECT: NavObj = {
  navLabel: CHILD_PAGE_NAVIGATION_LABEL,
  navURL: CHILD_URL,
};

export const USER_PAGE_NAVIGATION_OBJECT: NavObj = {
  navLabel: USER_PAGE_NAVIGATION_LABEL,
  navURL: USER_URL,
};
export const ROLE_PAGE_NAVIGATION_OBJECT: NavObj = {
  navLabel: ROLE_PAGE_NAVIGATION_LABEL,
  navURL: ROLE_URL,
};
export const TEAM_PAGE_NAVIGATION_OBJECT: NavObj = {
  navLabel: TEAM_PAGE_NAVIGATION_LABEL,
  navURL: TEAM_URL,
};
export const LOCATION_PAGE_NAVIGATION_OBJECT: NavObj = {
  navLabel: LOCATION_PAGE_NAVIGATION_LABEL,
  navURL: LOCATION_URL,
};

// collapse nav object
// client
export const CLIENT_COLLAPSE_NAVIGATION_OBJECT: NavCollapseObj = {
  navIcon: CLIENT_COLLAPSE_NAVIGATION_ICON,
  navLabel: CLIENT_COLLAPSE_NAVIGATION_LABEL,
};
export const REPORT_COLLAPSE_NAVIGATION_OBJECT: NavCollapseObj = {
  navIcon: REPORT_COLLAPSE_NAVIGATION_ICON,
  navLabel: REPORT_COLLAPSE_NAVIGATION_LABEL,
};
export const ADMIN_COLLAPSE_NAVIGATION_OBJECT: NavCollapseObj = {
  navIcon: ADMIN_COLLAPSE_NAVIGATION_ICON,
  navLabel: ADMIN_COLLAPSE_NAVIGATION_LABEL,
};

// nav module objects
export const CLIENT_NAVIGATION_MODULE_OBJECT: NavModuleObj = {
  childNavs: [
    CLIENT_PAGE_NAVIGATION_OBJECT,
    HOUSEHOLD_PAGE_NAVIGATION_OBJECT,
    ANC_PAGE_NAVIGATION_OBJECT,
    CHILD_PAGE_NAVIGATION_OBJECT,
  ],
  identifier: CLIENT_COLLAPSE_NAVIGATION_IDENTIFIER,
  parentNav: CLIENT_COLLAPSE_NAVIGATION_OBJECT,
};
export const REPORT_NAVIGATION_MODULE_OBJECT: NavModuleObj = {
  childNavs: [],
  identifier: REPORT_COLLAPSE_NAVIGATION_IDENTIFIER,
  parentNav: REPORT_COLLAPSE_NAVIGATION_OBJECT,
};
export const ADMIN_NAVIGATION_MODULE_OBJECT: NavModuleObj = {
  childNavs: [
    USER_PAGE_NAVIGATION_OBJECT,
    ROLE_PAGE_NAVIGATION_OBJECT,
    TEAM_PAGE_NAVIGATION_OBJECT,
    LOCATION_PAGE_NAVIGATION_OBJECT,
  ],
  identifier: ADMIN_COLLAPSE_NAVIGATION_IDENTIFIER,
  parentNav: ADMIN_COLLAPSE_NAVIGATION_OBJECT,
};
