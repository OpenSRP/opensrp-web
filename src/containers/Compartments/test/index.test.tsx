import reducerRegistry from '@onaio/redux-reducer-registry';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import ConnectedCompartments from '..';
import { PREGNANCY, PREGNANCY_REGISTRATION, TYPE_SMS_FILTER_FUNCTION } from '../../../constants';
import locationsReducer, {
  reducerName as locationsReducerName,
} from '../../../store/ducks/locations';
import smsReducer, {
  fetchSms,
  reducerName as smsReducerName,
  SmsData,
} from '../../../store/ducks/sms_events';
import store from '../../../store/index';
import { smsDataFixtures } from './fixtures';

const history = createBrowserHistory();
reducerRegistry.register(smsReducerName, smsReducer);
reducerRegistry.register(locationsReducerName, locationsReducer);

describe('Compartments', () => {
  it('must render without crashing', () => {
    shallow(
      <Provider store={store}>
        <ConnectedCompartments />
      </Provider>
    );
  });
  it('must render correctly', () => {
    store.dispatch(fetchSms([]));
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedCompartments
            filterArgs={
              [
                (smsData: SmsData) => {
                  return smsData.sms_type === PREGNANCY_REGISTRATION;
                },
              ] as TYPE_SMS_FILTER_FUNCTION[]
            }
            module={PREGNANCY}
          />
        </Router>
      </Provider>
    );

    // compartment title should have the correct text
    expect(toJson(wrapper.find('#compartment_title'))).toMatchSnapshot('compartment title');

    // the sub-heading right below the title
    expect(toJson(wrapper.find('#breadcrumb'))).toMatchSnapshot('subtitle');

    // ensure 3 data circle cards are found
    expect(wrapper.find('DataCircleCard')).toHaveLength(3);

    expect(wrapper.find('div.compartment-wrapper.compartments').find('Row')).toHaveLength(2);
  });

  it('must display the correct data', () => {
    store.dispatch(fetchSms(smsDataFixtures));
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedCompartments
            filterArgs={
              [
                (smsData: SmsData) => {
                  return smsData.sms_type === PREGNANCY_REGISTRATION;
                },
              ] as TYPE_SMS_FILTER_FUNCTION[]
            }
            module={PREGNANCY}
          />
        </Router>
      </Provider>
    );

    // the numbers in this snapshot encode the correctness of the private functions in the Compartments component
    expect(
      toJson(wrapper.find('div.compartment-wrapper.compartments').find('.dataCircleCard.card'))
    ).toMatchSnapshot('compartments');
  });
});
