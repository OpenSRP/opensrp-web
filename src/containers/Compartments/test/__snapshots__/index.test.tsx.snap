// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compartments must display the correct data: compartments 1`] = `
Array [
  <div
    className="dataCircleCard  card"
  >
    <CardTitle
      tag="div"
    >
      <div
        className="card-title"
      >
        22 Total Pregnancies
      </div>
    </CardTitle>
    <CardBody
      tag="div"
    >
      <div
        className="card-body"
      >
        <ul
          className="circlesRow"
        >
          <li
            className="red"
          >
            <Link
              onClick={[Function]}
              to="/pregnancy_compartments/hierarchicaldata/Pregnancy/high/22 Total Pregnancies"
            >
              <a
                href="/pregnancy_compartments/hierarchicaldata/Pregnancy/high/22 Total Pregnancies"
                onClick={[Function]}
              >
                <span
                  className="number"
                >
                  4
                </span>
              </a>
            </Link>
            <span
              className="risk-level"
            >
              High Risk
            </span>
          </li>
          <li
            className="orange"
          >
            <Link
              onClick={[Function]}
              to="/pregnancy_compartments/hierarchicaldata/Pregnancy/low/22 Total Pregnancies"
            >
              <a
                href="/pregnancy_compartments/hierarchicaldata/Pregnancy/low/22 Total Pregnancies"
                onClick={[Function]}
              >
                <span
                  className="number"
                >
                  2
                </span>
              </a>
            </Link>
            <span
              className="risk-level"
            >
              Low Risk
            </span>
          </li>
          <li
            className="green"
          >
            <Link
              onClick={[Function]}
              to="/pregnancy_compartments/hierarchicaldata/Pregnancy/no/22 Total Pregnancies"
            >
              <a
                href="/pregnancy_compartments/hierarchicaldata/Pregnancy/no/22 Total Pregnancies"
                onClick={[Function]}
              >
                <span
                  className="number"
                >
                  5
                </span>
              </a>
            </Link>
            <span
              className="risk-level"
            >
              No Risk
            </span>
          </li>
        </ul>
      </div>
    </CardBody>
  </div>,
  <div
    className="dataCircleCard  card"
  >
    <CardTitle
      tag="div"
    >
      <div
        className="card-title"
      >
        0 Total Pregnancies due in 2 weeks
      </div>
    </CardTitle>
    <CardBody
      tag="div"
    >
      <div
        className="card-body"
      >
        <ul
          className="circlesRow"
        >
          <li
            className="red"
          >
            <Link
              onClick={[Function]}
              to="/pregnancy_compartments/hierarchicaldata/Pregnancy/high/0 Total Pregnancies due in 2 weeks"
            >
              <a
                href="/pregnancy_compartments/hierarchicaldata/Pregnancy/high/0 Total Pregnancies due in 2 weeks"
                onClick={[Function]}
              >
                <span
                  className="number"
                >
                  0
                </span>
              </a>
            </Link>
            <span
              className="risk-level"
            >
              High Risk
            </span>
          </li>
          <li
            className="orange"
          >
            <Link
              onClick={[Function]}
              to="/pregnancy_compartments/hierarchicaldata/Pregnancy/low/0 Total Pregnancies due in 2 weeks"
            >
              <a
                href="/pregnancy_compartments/hierarchicaldata/Pregnancy/low/0 Total Pregnancies due in 2 weeks"
                onClick={[Function]}
              >
                <span
                  className="number"
                >
                  0
                </span>
              </a>
            </Link>
            <span
              className="risk-level"
            >
              Low Risk
            </span>
          </li>
          <li
            className="green"
          >
            <Link
              onClick={[Function]}
              to="/pregnancy_compartments/hierarchicaldata/Pregnancy/no/0 Total Pregnancies due in 2 weeks"
            >
              <a
                href="/pregnancy_compartments/hierarchicaldata/Pregnancy/no/0 Total Pregnancies due in 2 weeks"
                onClick={[Function]}
              >
                <span
                  className="number"
                >
                  0
                </span>
              </a>
            </Link>
            <span
              className="risk-level"
            >
              No Risk
            </span>
          </li>
        </ul>
      </div>
    </CardBody>
  </div>,
  <div
    className="dataCircleCard  card"
  >
    <CardTitle
      tag="div"
    >
      <div
        className="card-title"
      >
        0 Total Pregnancies due in 1 week
      </div>
    </CardTitle>
    <CardBody
      tag="div"
    >
      <div
        className="card-body"
      >
        <ul
          className="circlesRow"
        >
          <li
            className="red"
          >
            <Link
              onClick={[Function]}
              to="/pregnancy_compartments/hierarchicaldata/Pregnancy/high/0 Total Pregnancies due in 1 week"
            >
              <a
                href="/pregnancy_compartments/hierarchicaldata/Pregnancy/high/0 Total Pregnancies due in 1 week"
                onClick={[Function]}
              >
                <span
                  className="number"
                >
                  0
                </span>
              </a>
            </Link>
            <span
              className="risk-level"
            >
              High Risk
            </span>
          </li>
          <li
            className="orange"
          >
            <Link
              onClick={[Function]}
              to="/pregnancy_compartments/hierarchicaldata/Pregnancy/low/0 Total Pregnancies due in 1 week"
            >
              <a
                href="/pregnancy_compartments/hierarchicaldata/Pregnancy/low/0 Total Pregnancies due in 1 week"
                onClick={[Function]}
              >
                <span
                  className="number"
                >
                  0
                </span>
              </a>
            </Link>
            <span
              className="risk-level"
            >
              Low Risk
            </span>
          </li>
          <li
            className="green"
          >
            <Link
              onClick={[Function]}
              to="/pregnancy_compartments/hierarchicaldata/Pregnancy/no/0 Total Pregnancies due in 1 week"
            >
              <a
                href="/pregnancy_compartments/hierarchicaldata/Pregnancy/no/0 Total Pregnancies due in 1 week"
                onClick={[Function]}
              >
                <span
                  className="number"
                >
                  0
                </span>
              </a>
            </Link>
            <span
              className="risk-level"
            >
              No Risk
            </span>
          </li>
        </ul>
      </div>
    </CardBody>
  </div>,
]
`;

exports[`Compartments must render correctly: compartment title 1`] = `
<h2
  id="compartment_title"
>
  Compartments
</h2>
`;

exports[`Compartments must render correctly: subtitle 1`] = `
<p
  id="breadcrumb"
>
  <span
    id="breadcrumb-span"
  />
</p>
`;
